woocommerce/woocommerce-ios
# Shipping Labels: Customs form validation for item details

Part of #4687
This PR is based on #4806, so please make sure that PR is reviewed first.

# Description
This PR adds validation for item details in customs form of Shipping Labels flow.

# Changes
1. Updated reusable view / view modifier:
- New modifier `ErrorStyle` for text with `.body` font and red color for errors.
- New view `ValidationErrorRow` to display validation error.
2. Updated view model for item details to add validations for item description, value, weight, HS tariff number, origin country.
3. Updated item details input view to add validation error rows when validation fails.

# Demo
![item-details-validation](https://user-images.githubusercontent.com/5533851/129563005-e573ff95-405c-4d81-a394-8ddbf39e4b19.gif)

# Testing steps
1. Make sure that your test store is eligible for creating shipping labels (has WCShip plugin installed).
2. Open Orders tab and select an order with international shipping (origin country different from destination country).
3. Select Create Shipping Label.
4. Skip through Ship From, Ship To, Package Detail to configure Customs.
5. Notice that a validation error appears when either description / value / weight is empty.
6. Notice that a validation error appears when tariff number is entered but the length is less than 6. You can also try pasting some random text in this field and see that the error says the number is invalid.
7. Notice that when a validation fails, there's a red notice icon shown in the header of the item detail input form.

Update release notes:
